src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -305,6 +306,27 @@ public boolean apply(StepDescriptor i) {
         return r;
     }
 
+    /** Return true if we can easily create a nice String for user display from the object */
+    static boolean isAbleToUseToStringForDisplay(@CheckForNull Object o) {
+        return (o != null && (o instanceof Comparable || o instanceof CharSequence)); // Covers our base types
```

Comment:
@stephenc There is no way to determine whether a toString method actually returns something useful to the user (even checking for overrides is no guarantee).  Comparable and CharSequence are not *perfect* but they cover 90% of cases, and a less-than-useful result here only results in a bit of visual ugliness.  In such cases we can always have the StepDescriptor provide its own implementation.     Number, Boolean, and most classes with a useful toString tend to implement Comparable.

  Initially I explored what you are suggesting, and them realized how many cases would be rejected if you only whitelisted a handful of types.  You also forgot Enums, for example, and some other Jenkins objects potentially.  I can guarantee there are others.  

CharSequence allows for other serializable character types besides Strings.  

Instanceof *does* cover the null case, but putting that check up front short-circuits a complex chain of conditionals.